### PR TITLE
Add target selector

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

image

https://github.com/home-assistant/core/pull/43831

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-area-picker.ts [hunk 1]
```diff
@@ -71,39 +82,225 @@ export class HaAreaPicker extends SubscribeMixin(LitElement) {
...
+        for (const entity of entities) {
+          if (!entity.device_id) {
+            continue;
+          }
+          if (!(entity.device_id in deviceEntityLookup)) {
+            deviceEntityLookup[entity.device_id] = [];
+          }
+          deviceEntityLookup[entity.device_id].push(entity);
+        }
+        inputDevices = [...devices];
```
Comment (balloob) [inline src/components/ha-area-picker.ts:176]: Why would we make a copy here? We don't ever mutate the array do we?

File: src/components/ha-target-picker.ts [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+          return this._renderChip(
+            "entity_id",
+            entity_id,
+            entity ? computeStateName(entity) : entity_id,
+            entity ? stateIcon(entity) : undefined
+          );
+        })}
+      </div>
+      ${this._renderPicker()}
+      <div class="mdc-chip-set">
```
Comment (balloob) [inline src/components/ha-target-picker.ts:154]: In a future PR, should we make this a component ?

File: src/components/ha-target-picker.ts [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+                tabindex="-1"
+                role="button"
+                .label=${"Expand"}
+                .id=${id}
+                .type=${type}
+                @click=${this._handleExpand}
+              >
+                <ha-svg-icon .path=${mdiUnfoldMoreVertical}></ha-svg-icon>
+              </mwc-icon-button>
+              <paper-tooltip animation-delay="0"
```
Comment (balloob) [inline src/components/ha-target-picker.ts:271]: Let's add some CSS to define a width, felt a little cramped.

File: src/translations/en.json [hunk 4]
```diff
@@ -334,6 +334,16 @@
...
       },
+      "target-picker": {
+        "expand_area_id": "Expand this area in the seperate device and entities that it contains. It will update when the area changes.",
+        "expand_device_id": "Expand this device in seperate entities.",
+        "remove_area_id": "Remove area",
+        "remove_device_id": "Remove device",
+        "remove_entity_id": "Remove entity",
+        "add_area_id": "Select area",
+        "add_device_id": "Select device",
+        "add_entity_id": "Select entity"
```
Comment (balloob) [inline src/translations/en.json:345]: Little shorter:

```suggestion
        "add_area_id": "Pick area",
        "add_device_id": "Pick device",
        "add_entity_id": "Pick entity"
```

**Review by balloob — APPROVED** [review]

File: src/components/ha-target-picker.ts (hunk 2 above)
Comment (bramkragten) [reply src/components/ha-target-picker.ts:154]: I'm waiting for `mwc-chip-set`, but they delayed that again...

File: src/components/ha-target-picker.ts (hunk 3 above)
Comment (bramkragten) [reply src/components/ha-target-picker.ts:271]: image

**Pull request merged**